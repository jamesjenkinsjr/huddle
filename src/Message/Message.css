p.portal__message-content {
  font-size: 16px;
  word-wrap: break-word;
}

p.portal__message-info {
  font-size: 10px;
}

@media screen and (min-width: 1024px) {
  p.portal__message-content {
    font-size: 18px;
  }

  p.portal__message-info {
    font-size: 12px;
  }
}