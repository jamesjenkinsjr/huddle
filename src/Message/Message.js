import React from 'react'
import './Message.css'

export default function Message(props) {
  return (
    <li className='portal__message-item'>
      <p>{props.content}</p>
      <p>Sent {props.create_timestamp} - {props.author} </p>
    </li>
  );
}