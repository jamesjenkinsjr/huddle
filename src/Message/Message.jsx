import React from 'react'
import './Message.css'

export default function Message(props) {
  return (
    <li className="portal__message-item">
      <p className="portal__message-content">{props.content}</p>
      <p className="portal__message-info">
        Sent {props.create_timestamp} - {props.author}
      </p>
    </li>
  )
}
