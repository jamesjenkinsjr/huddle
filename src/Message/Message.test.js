import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Message from './Message'

describe('Message component', () => {
  it('renders Message by default', () => {
    const wrapper = shallow(<Message />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('renders given props', () => {
    const props = {
      author: 'test',
      content: 'This is some test content.',
      create_timestamp: new Date('2019-10-23T16:09:09.303Z').toLocaleString(),
      portal_id: 'a7580f03-d358-4f3f-a5f8-6b69ba02d83a'
    }
    console.log(props.create_timestamp)
    const wrapper = shallow(<Message {...props} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
