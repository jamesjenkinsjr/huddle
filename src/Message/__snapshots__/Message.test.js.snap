// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders Message by default 1`] = `
<li
  className="portal__message-item"
>
  <p
    className="portal__message-content"
  />
  <p
    className="portal__message-info"
  >
    Sent 
     - 
  </p>
</li>
`;

exports[`Message component renders given props 1`] = `
<li
  className="portal__message-item"
>
  <p
    className="portal__message-content"
  >
    This is some test content.
  </p>
  <p
    className="portal__message-info"
  >
    Sent 
    10/23/2019, 12:09:09 PM
     - 
    test
  </p>
</li>
`;
